app/assets/javascripts/ui_components/select.js.jsx

```diff
@@ -0,0 +1,144 @@
+(function () {
+  var strategies = {
+    single: {
+      initialValue: function(selected) {
+        return selected || '';
+      },
+
+      updateValue: function(current, update) {
+        return update ? update.selected : null;
+      },
+
+      updateOptions: function(existingOptions, newOptions) {
+        return newOptions;
+      }
+    },
+
+    multiple: {
+      initialValue: function(selected) {
+        return selected ? _.flatten([selected]) : [];
+      },
+
+      updateValue: function(current, update) {
+        if (update.selected) {
+          return current.concat(update.selected);
+        } else if (update.deselected) {
+          return _.without(current, update.deselected);
+        }
+      },
+
+      updateOptions: function(existingOptions, newOptions) {
+        return _.union(existingOptions, newOptions);
+      }
+    }
+  };
+
+
+  var Select = React.createClass({
+    chosenDefaults: {
+      searchContains: true
+    },
+
+    propTypes: {
+      remote_options: React.PropTypes.string,
+      multiple: React.PropTypes.bool,
+      options: React.PropTypes.any,
+      name: React.PropTypes.string,
+      selected: React.PropTypes.oneOfType([
+        React.PropTypes.string,
+        React.PropTypes.arrayOf(React.PropTypes.string)
+      ]),
+      placeholder: React.PropTypes.string,
+      classes: React.PropTypes.string
+    },
+
+    getInitialState: function() {
+      return { value: this.strategy().initialValue(this.props.selected), search: '' };
+    },
+
+    componentDidMount: function() {
+      if (this.props.remote_options) {
+        this.getChosenInput().on('keyup', _.bind(function(e) {
```

Comment:
JSX already has support for ES6-style arrow functions, so no need for `bind(this, ...)` anymore.